### PR TITLE
added task need to add the api url correct and fix the appstyle manager

### PR DESCRIPTION
**Pull request by Allan-Nava**

### CODE REVIEW

**Comment by Allan-Nava** [comment]
It miss the correct path for api and configuration for the `config.endpoint` @georgejecook

File: samples/src/source/screens/TaskScreenVM.bs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import "pkg:/source/roku_modules/maestro/mx/BaseViewModel.brs"
+import "pkg:/source/roku_modules/maestro/mx/VMTransformFunctions.brs"
+'
+namespace screens
+  class TaskScreenVM extends mx.BaseViewModel
+
+    public focusedIndex = -1
```
Comment (georgejecook) [inline samples/src/source/screens/TaskScreenVM.bs:7]: thisis not the code formatting of the project. please don't use this formatting.

File: samples/src/source/screens/TaskScreenVM.bs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public items        = createObject("roSGNode", "ContentNode")
+    public hasItems     = false
+    public loadDataTask
```
Comment (georgejecook) [inline samples/src/source/screens/TaskScreenVM.bs:10]: bare in mind that the correct way to declare this is with types. maestro has type polyfills (see Types-Polyfill.d.brs)
`public loadDataTas ac mc.types.node`

File: samples/src/components/screens/TaskScreen.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<component
+    name='TaskScreen'
+    extends='mv_BaseScreen'
+    vm='screens.TaskScreenVM'>
+    <interface>
+	</interface>
+
+    <children>
```
Comment (georgejecook) [inline samples/src/components/screens/TaskScreen.xml:9]: Can we get some kind of visual representation that the data is loading?
 - like a button we click to load?
 - a grid that updats when the data is loaded?

File: samples/src/source/screens/TaskScreenVM.bs (hunk 1 above)
Comment (Allan-Nava) [reply samples/src/source/screens/TaskScreenVM.bs:7]: How? I just wanna add a screen for the task example 

Thanks
Allan

File: samples/src/source/tasks/ContentTask.bs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+@task("ContentTask", "Task")
+@strict
+class ContentTask
+  private log = new log.Logger("ContentTask")
+
+  @inject("styleManager")
+  private styleManager
+
+  ' @injectClass("mc.net.Request")
```
Comment (georgejecook) [inline samples/src/source/tasks/ContentTask.bs:14]: why is this commented out? was there a problem with this? - I think it's more correct to instnatiate it, as it goes; but wondering - if this code is not used, remove it.

File: samples/src/source/tasks/ContentTask.bs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  private request = new mc.net.Request()
+
+  private config = mc.getPath(m.styleManager@.getStyle("app"), "server", {})
+
+  private function execute(args as mc.types.assocarray)
+    command = lcase(mc.getPath(args, "command", ""))
+    result = { isOK: false }
+    ' need to integrate the api stuff
+    result = m.request.get(`${mc.config.endpoint}/movies.json`)
```
Comment (georgejecook) [inline samples/src/source/tasks/ContentTask.bs:23]: I don't understand this line. what is `mc.config.endpoint`?  - surely this is a compile error, right?

File: samples/src/source/tasks/ContentTask.bs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+class ContentTask
+  private log = new log.Logger("ContentTask")
+
+  @inject("styleManager")
+  private styleManager
+
+  ' @injectClass("mc.net.Request")
+  private request = new mc.net.Request()
+
+  private config = mc.getPath(m.styleManager@.getStyle("app"), "server", {})
```
Comment (georgejecook) [inline samples/src/source/tasks/ContentTask.bs:17]: this does not need a sytle manager? why is this here?

File: samples/src/source/tasks/ContentTask.bs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+  private function execute(args as mc.types.assocarray)
+    command = lcase(mc.getPath(args, "command", ""))
```
Comment (georgejecook) [inline samples/src/source/tasks/ContentTask.bs:20]: I appreciate you making this example; but can we make it more useful for other people. Would you please actually load some data and parse it - nothing special, get some items from some json dump somewhere and create a few nodes to show in a rowlist.. - just grabbing the title will be enough.
like: https://jsonplaceholder.typicode.com/photos

**Review by georgejecook — CHANGES_REQUESTED** [review]
Appreciate that you are makiing the effort to make an example; but it needs more utility. please see my comments. 

Thanks!

**Comment by Allan-Nava** [comment]
> Appreciate that you are makiing the effort to make an example; but it needs more utility. please see my comments.
> 
> Thanks!

You are right!

File: samples/src/source/tasks/ContentTask.bs (hunk 7 above)
Comment (Allan-Nava) [reply samples/src/source/tasks/ContentTask.bs:20]: > I appreciate you making this example; but can we make it more useful for other people. Would you please actually load some data and parse it - nothing special, get some items from some json dump somewhere and create a few nodes to show in a rowlist.. - just grabbing the title will be enough.
> like: https://jsonplaceholder.typicode.com/photos

I will do it!

File: samples/src/source/tasks/ContentTask.bs (hunk 5 above)
Comment (Allan-Nava) [reply samples/src/source/tasks/ContentTask.bs:23]: Of course in fact I don't know how to implement it

**Review by Allan-Nava — COMMENTED** [review]
Fix it

**Comment by georgejecook** [comment]
thanks for your contrinbution; but I'm moving things in a different direction right now. closing this.
